### ISSUE TITLE
Bring your own Cert for Applications

### ISSUE THREAD

**Issue by jimmykarily** [epinio/epinio#63]
Depends on: https://github.com/SUSE/carrier/issues/62

We should implement the automated TLS setup for applications but we should also allow the usage of existing certificates. Some users may already be paying for their certificates and they may want to use them.

After we fix #62 we should find a way to allow users to use existing certificates instead of letting Carrier create new ones for them.

**Comment by jimmykarily** [comment]
@agracey suggested that we could implement "bring your own cert" as "bring your own issuer". In other words, allow the users to specify another issues (their own issuer) that serves their own certificates for various domains (at some point we will support multiple domains). If the domain in use is not known to the user's issuer it could fallback to let's encrypt.

We can have a spike to investigate if that is possible.

**Comment by agracey** [comment]
Is this done with https://github.com/epinio/epinio/issues/564? Or is it separate?

**Comment by jimmykarily** [comment]
No this is separate. This is mentioned briefly in #384 but maybe deserves another card (or is this one that card?). What we want it the user to be able to use any value for the `--tls-issuer` flag of the install command. This way, they can create an issuer with a custom CA they own. Optionally, epinio will provide a command to help with that.

**Comment by jimmykarily** [comment]
Specifying an arbitrary string as the issuer name is already supported (@andreas-kupries ?). What's left is:

- Verify that arbitrary issuer names are supported
- Document how to create an issuer with an existing ca cert
- When installing epinio, validate that the given issuer exists before trying to install. If for example the user makes a typo, this shouldn't result in a broken installation (should just print an error and exit before anything is installed).
- [optional] auto completion for existing issuers on the `epinio install` command.

**Comment by manno** [comment]
Epinio now supports setting a custom cert-manager issuers when installing. 

https://github.com/epinio/epinio/blob/d9562397e69a4c0f47351740aff0b0154ba60df6/docs/user/howtos/certificate_issuers.md


We can open another issue, if we want to revisit custom certificates (not created by a CA/letsencrypt, but e.g. bought online) for  pushed apps.
